ext/nnstreamer/tensor_filter/tensor_filter_snpe.cc

```diff
@@ -221,17 +220,15 @@ snpe_subplugin::runtimeToString (zdl::DlSystem::Runtime_t runtime)
  * @brief Internal method to get names of output layers from tensors information.
  */
 bool
-snpe_subplugin::set_output_layer_names (const GstTensorsInfo *info)
+snpe_subplugin::set_output_tensor_names (const GstTensorsInfo *info)
 {
   for (unsigned int i = 0; i < info->num_tensors; ++i) {
     if (info->info[i].name == nullptr) {
       /* failed */
-      nns_loge ("Given layer name with index %u is invalid.", i);
-      output_layer_names.clear ();
+      nns_loge ("Given tensor name with index %u is invalid.", i);
```

Comment:
```s/tensor/output tensor/```
```s/invalid/invalid, it is null/```
?

And.. do  you need to check if ```info->info[i].name[0] == '\0'``` as well?

Reply:
If it is failed at ```i > 0```, don't you need to free elements of output_tensor_names_list ?

Reply:
> `s/tensor/output tensor/` `s/invalid/invalid, it is null/` ?
> 
> And.. do you need to check if `info->info[i].name[0] == '\0'` as well?

Updated, Thanks!



> If it is failed at `i > 0`, don't you need to free elements of output_tensor_names_list ?

The SNPE SDK does not provide API for freeing elements or cleaning. I updated the `cleanup ()` function to reset the list. PTAL